packages/authentication/src/authenticate.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import 'reflect-metadata';
+
+export class AuthenticationMetadata {
+  strategyName: string;
+  options: string[];
+
+  constructor(private strategy: string, private optionValues?: string[]) {
+    this.strategyName = strategy;
+    this.options = optionValues || [];
+  }
+  getMetadata() {
+    return {strategy: this.strategyName, options: this.options};
+  }
+}
+
+export function authenticate(strategyName: string, options? : string[]) {
+  return function(controllerClass: Object, methodName: string) {
+    const metadata: AuthenticationMetadata = new AuthenticationMetadata(strategyName, options || []);
+    Reflect.defineMetadata('loopback:authenticate', metadata.getMetadata(), controllerClass, methodName);
+  };
+}
+
+export function getAuthenticateMetadata(controllerObj: Object, methodName: string): boolean {
+  return Reflect.getMetadata('loopback:authenticate', controllerObj, methodName) || undefined;
```

Comment:
Weird. We do have `strictNullChecks` enabled, I am surprised the compiler does not complain.

As far as I am concerned, returning `boolean | undefined` is fine for now.